python/ray/serve/pipeline/deployment_method_node.py

```diff
@@ -29,13 +29,13 @@ def __init__(
         # Serve handle is sync by default.
         if (
             "sync_handle" in self._bound_other_args_to_resolve
-            and self._bound_other_args_to_resolve.get("sync_handle") is True
+            and self._bound_other_args_to_resolve.get("sync_handle") is False
```

Comment:
should we explicitly be handling the `None` case too?

Reply:
sure, let me enhance this a bit. I structured it to be very specific and picky on the async handle case, where sync handle is the catch-all for everything else.